lib/src/license_detection/new_tokenizer.dart

```diff
@@ -0,0 +1,278 @@
+      continue;
+    }
+
+    firstInLine = false;
+
+    final text = _cleanToken(token.normalizedText);
+
+    output.add(NewToken(text, tokenID++, token.line, token.fileSpan));
+  }
+
+  return output;
+}
+
+/// Normalizes the tokens using the approach of [google licenseClassifier]
+/// to provide better chances of matching.
+/// [google licenseClassifier]  https://github.com/google/licenseclassifier/blob/bb04aff29e72e636ba260ec61150c6e15f111d7e/v2/tokenizer.go#L85
+String _cleanToken(String tok) {
+  final runes = tok.runes;
+  var buffer = StringBuffer();
+
+  if (!_isLetter(runes.first)) {
+    if (_isDigit(runes.first)) {
+      runes.forEach((rune) {
+        if (_isDigit(rune) || rune == _dot || rune == _hiphen) {
+          buffer.write(String.fromCharCode(rune));
+        }
+      });
+
+      var text = buffer.toString();
+
+      if (text.runes.last == _dot) {
+        text = text.substring(0, text.length);
+      }
+
+      return text;
+    }
+  }
+
+  runes.forEach((rune) {
+    if (_isLetter(rune)) {
+      buffer.write(String.fromCharCode(rune));
+    }
+  });
+
+  var text = buffer.toString();
+
+  // If this is a varietal word normalize it according to
+  // Guideline 8.1.1: Legally equal words must be treated same.
+  text = _varietalWords[text] ?? text;
+  return text;
+}
+
+bool _isListItem(String token) {
+  final end = token[token.length - 1];
+  final start = token.substring(0, token.length - 1);
+
+  if ((end == '.' || end == ')') && _headers.contains(start)) {
+    return true;
+  }
+
+  if (_numberHeaderRe.hasMatch(token)) {
+    return true;
+  }
+
+  return false;
+}
+
+bool _isDigit(int rune) {
+  return (rune > 47 && rune < 58);
+}
+
+bool _isLetter(int rune) {
+  return ((rune > 96 && rune < 123) || (rune > 64 && rune < 91));
+}
+
+bool _isSpace(int rune) {
+  return rune == 32;
+}
+
+bool _isAlphaNumeric(int rune) {
+  return _isDigit(rune) || _isLetter(rune);
+}
+
+bool _isNewLine(int char) {
+  switch (char) {
+    case 10:
+    case 12:
+    case 133:
+      return true;
+  }
+  return false;
+}
+
+final _headers = HashSet.from(
+    'q w e r t y u i o p a s d f g h j k l z x c v b n m i ii iii iv vi vii ix xi xii'
+        .split(' '));
+
+final _numberHeaderRe = RegExp(r'^\d{1,2}(\.\d{1,2})*[\.)]$');
+
+final _horizontalWhiteSpaceRegex = RegExp(r'[^\S\r\n]');
+
+final Map<RegExp, String> _equivalentPunctuationMarks = {
+  // Guideline 5.1.2: All variants of dashes considered equivalent.
+  RegExp(r'[-֊־᠆‐‑–—﹘﹣－‒⁃⁻⎯─⏤]'): '-',
+
+  // Guideline Guideline 5.1.3: All variants of quotations considered equivalent.
+  RegExp(r'[“‟”’"‘‛❛❜〝〞«»‹›❝❞]'): "'",
+};
+
+const int _dot = 46;
+const int _hiphen = 45;
+
+/// Words obtained from [SPDX corpus][]
+///
+/// [SPDX corpus]: https://github.com/spdx/license-list-XML/blob/master/equivalentwords.txt
+// TODO : Some words are left out from the original list
+// fina a way to work with remaining words.
+final HashMap<String, String> _varietalWords = HashMap.from({
```

Comment:
Using `Map<String,String>` is usually fine.. Unless you have a specific reason to what a specific map implementation, like a hash-map, don't specify it and assume that the JIT/compiler/runtime will figure out what to use.

```suggestion
final _varietalWords = <String, String>{
```